refactor(ast): check whether there is a single `quasi` in `TemplateLiteral::quasi`

Dunqing opened the pull request:
As the changes show, `TemplateLiteral::quasi` is always paired with a check to use, that check is whether there is only one single `quasi` in this `TemplateLiteral`. This PR is to put this check into the `TemplateLiteral::quasi` API, so that we can simplify the use of `TemplateLiteral::quasi`

overlookmotel reviewed (approved):
Much better!

overlookmotel commented:
### Merge activity

* **Jul 14, 7:19 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
refactor(ast): check whether there is a single `quasi` in `TemplateLiteral::quasi` (#12265)

As the changes show, `TemplateLiteral::quasi` is always paired with a check to use, that check is whether there is only one single `quasi` in this `TemplateLiteral`. This PR is to put this check into the `TemplateLiteral::quasi` API, so that we can simplify the use of `TemplateLiteral::quasi`
